app/controllers/application_controller.rb

```diff
@@ -9,4 +9,8 @@ class ApplicationController < ActionController::Base
     def current_user
       @current_user ||= User.find(session[:user_id])
     end
+
+    def jwt_encode(payload)
+      JWT.encode(payload, Rails.application.secrets.secret_key_base)
```

Comment:
Why do we encode here if all the other jwt logic is in api_controller?

Reply:
We encode here because this is where a user enters the app. In most situations a user would call the API and get a token back, but in this case it's not really necessary since they're already authenticated. I just pass this token on to React as a prop when they're routed to a dashboard.